src/architecture/msgPayloadDefC/TypesTestMsgPayload.h

```diff
@@ -0,0 +1,66 @@
+/*
+ ISC License
+
+ Copyright (c) 2024, Astroscale Japan
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef TYPES_TEST_MESSAGE_H
+#define TYPES_TEST_MESSAGE_H
+
+#include <stdint.h>
+
+#define TYPES_TEST_ARRAY_SIZE 3
+
+/*! @brief Structure used to test the handling of various built-in types in message payloads across SWIG C/Python bridge */
+typedef struct {
+    // Scalars
+    int8_t i8Test;
```

Comment:
All these variables must have doxygen comment strings starting with `//!<`.  Right now building the documentation throws a lot of doxygen warnings that all these variables are not defined.

Reply:
Done

Reply:
I comment here only because it's relevant to the requested change: I don't think we should be adding documenting comments to unit tests (integration tests and examples are a different consideration as they serve two purposes; testing and demonstration). Rather, we should exclude tests from the documentation generation process. Unit tests (see my previous distinction) are not user facing and therefore do not bare documenting. If a test warrants further description/context then that can be added in the normal doc string style above the test. Happy to add more information about this rationale if desired.

Reply:
@patkenneally I agree with that reasoning. How would I exclude it from the doc generation?

Reply:
To continue the discussion on including this msg definition file in the documentation.  As it is present inside this msg definition folder, I can see users wondering what it is for.  To me this means the file needs to contain good documentation comments inside the *.h for sure explaining in clear terms what its purpose is.  

The question is then does this file have to be displayed in the BSK HTML documentation.  We do include the module unit tests inside our documentation as they do serve as an example on how to create and run a particular BSK module.

In this situation there is no module, but there is a unit test that gets run.  If we exclude the test msg file from documentation, I feel we would also need to exclude the associated python test file from documentation to be consistent.  At first blush, however, I find it useful to have HTML access to all the test scripts being run inside the `src/architecture` folder.   I do hope to have many more of these in the future as we revise and improve core features of BSK and include test files.  

I'll think on this some more.  Just trying to writing down my thoughts as I can see arguments for both sides here.

Reply:
@joaogvcarneiro , we are not going to a format where code doesn't have variable definition.  There is no way to automate documentation and excluded "obvious" names or not.

Reply:
Just to ensure we're all understanding @joaogvcarneiro 's comment: in the main about commenting he is correct. The comment does not suggest to move away from adding documentation of variables, rather it calls into question the utility of doing so for some code. And it is a point well made.

> Even if this msg definition is not included in the BSK documentation, the variables used within should be documented.
Agreed, and in this case the documentation/commenting should be to aid understanding, not for doxygen's consumption. The file and the test should be excluded from documentation generation. 

Ideally, if our build system was appropriately flexible, the test message file would be located along side the test code and not with the rest of the message definitions. Thus appropriately separating test code from production code.

@schaubh can you be specific about what you find useful in having the HTML documentation for a test file? Also what of the C/C++ tests? Prevailing wisdom around tests is that tests are explicit, clear, simple, and self documenting. As a result the test code itself is the best documentation and anything further is likely to be anemic in regards the information it provides.

Reply:
I do find my self often jumping from the module documentation to the test scripts to see what is going on.  I know there is at least one heavy user of this as I look at BSK code and scripts.

Reply:
I hear that and this makes sense for example scenarios and integrated tests. However this test, among others, doesn't demonstrate anything user facing. If there is, this test is not helpful in demonstrating that.

Reply:
You asked me about accessing test files in general, I definitely see utility for that.  This one is different is it test a core capability of BSK.  This could become user facing if someone is looking into how swig handles different integer types, for example.  But I agree this is a less likely scenario.  

If we want to start to exclude some test files and definitions, we should ensure we are consistent on this, not just for one file.  I'm sure we all agree on that ;-)

Reply:
Regarding excluding certain files from sphinx, I'm sure we can do that with some effort on the python level.  Excluding specific files from doxygen is another issue I would have to read up on first.  To the best of my knowledge sphinx calls doxygen, but if we tell sphinx not to include some files then this might work quite easily.  I'd have to test out some ideas on how to do this elegantly.

Reply:
Been thinking about this more last night, documentation of fundamental BSK scripts and tests of such features, versus focusing documentation just on user facing BSK modules and scenarios. We have always contained a Developer tab where we include a range of information on how to build modules, etc.  The audience of this is not just the user looking up modules, but also developer related items.  I know the developer documentation still is very much work in progress.  For example, we document how to build BSK modules, but don't discuss BSK effectors yet.  As we have been correcting BSK fundamental features like deprecation methods, etc., we have been adding tests that show up in our documentation inside the architecture tab.  I don't find any reason to stop including such information in our HTML documentation .

Reply:
@patkenneally @schaubh

Following up on this, is there some agreement on what to do for documentation?

Reply:
Yes, I need the BSK documentation to build without warnings.  Please add doxygen comments for the message variables to provide brief descriptions.  Thanks for checking.